<?xml version="1.0" encoding="UTF-8"?>
<launch>
	<arg name="sigverse_ros_bridge_port"    default="50001" />
	<arg name="ros_bridge_port"             default="9090" />
	<arg name="camera_ns"                   default="/camera" />
	<arg name="floor_z"                     default="-0.5"/>
	
	<node pkg="sigverse_turtlebot3" exec="recognize_pointed_direction" output="screen">
	    <param name="floor_z" value="$(var floor_z)"/>
		<!--
		<remap from="/tb3/joint_trajectory"       to="/tb3/joint_trajectory" />
		<remap from="/tb3/debug_markers"          to="/tb3/debug_markers" />
		<remap from="/tb3/instruction"            to="/tb3/instruction" />
		<remap from="/yolo_human_pose/detections_3d" to="/yolo_human_pose/detections_3d" />
		-->
	</node>

	<node pkg="sigverse_ros_bridge" exec="sigverse_ros_bridge" args="$(var sigverse_ros_bridge_port)"/>

	<include file="$(find-pkg-share rosbridge_server)/launch/rosbridge_websocket_launch.xml" > 
		<arg name="port" value="$(var ros_bridge_port)"/>
		<arg name="default_call_service_timeout" value="5.0"/>
		<arg name="call_services_in_new_thread" value="true"/>
		<arg name="send_action_goals_in_new_thread" value="true"/>
		<!-- <arg name="bson_only_mode" value="true"/> -->
	</include>

	<!-- YOLO Human Pose -->
	<include file="$(find-pkg-share yolo_bringup)/launch/yolov11.launch.py">
		<arg name="namespace" value="yolo_human_pose"/>
		<arg name="model" value="$(env HOME)/yolo_weights/yolo11n-pose.pt"/>
		<arg name="device" value="cpu"/>
		<arg name="threshold" value="0.8"/>
		<arg name="imgsz_height" value="376"/> 
		<arg name="imgsz_width" value="672"/>
		<!--
		<arg name="imgsz_height" value="188"/> 
		<arg name="imgsz_width" value="336"/>
		-->
		<arg name="input_image_topic" value="$(var camera_ns)/rgb/image_raw"/>
		<arg name="input_depth_topic" value="$(var camera_ns)/depth/image_raw"/>
		<arg name="input_depth_info_topic" value="$(var camera_ns)/depth/camera_info"/>
		<arg name="target_frame" value="base_link"/>
		<arg name="depth_image_units_divisor" value="1"/>
		<arg name="use_3d" value="True"/>
		<arg name="use_debug" value="True"/>
	</include>

	<node pkg="rviz2" exec="rviz2" args="-d $(find-pkg-share sigverse_turtlebot3)/launch/recognize_pointed_direction.rviz"/>
</launch>

